provider/pkg/await/await.go

```diff
@@ -718,19 +733,22 @@ func Deletion(c DeleteConfig) error {
 	}
 
 	// Obtain client for the resource being deleted.
-	client, err := c.ClientSet.ResourceClientForObject(c.Inputs)
+	client, err := c.ClientSet.ResourceClientForObject(c.Outputs)
```

Comment:
As tempting as it is to use `c.Inputs` here, it would change the status quo and create some code smell because the client is used with the `c.Outputs` in many places below.